sync/rpc/server.go

```diff
@@ -299,18 +359,44 @@ func (*SyncAgentServer) BackupRestore(ctx context.Context, req *BackupRestoreReq
 		return nil, err
 	}
 
+	// Successfully finish incremental full restore
+	if strings.HasSuffix(req.SnapshotFileName, ".snap_tmp") {
+		if backupName, err := backupstore.GetBackupFromBackupURL(util.UnescapeURL(req.Backup)); err != nil {
+			return nil, err
+		} else if err := s.FinishRestore(backupName); err != nil {
```

Comment:
Hmm, even we failed the restore we are still going to set the FinishRestore with backupName?

Reply:
When the RestoreBackup function starts running in the context of Sync agent(instead of backup process), this issue is addressed then.

Reply:
I don't understand..

Reply:
In case if the restore failed, only the isRestoring flag should be turned back to false but the lastRestored flag should not be updated in this scenario. Is that right?

Or are you saying that this code will be hit, even if the restore is failed? 
Then what I was trying to say is that - In that scenario, when we move add the `DoRestoreBackup` function in the next commit, that will make sure this code is not hit in case of failure.

Reply:
OK.

Also, don't use `else` when you suppose to execute the next line of code when the previous code succeed.